### PR TITLE
docs(DHT): update dht README

### PR DESCRIPTION
**Pull request by juslesan**
## Summary

Updated the DHT packages README.md file. Added a general summary of the DHT's feature set. The update includes help for some common configuration pitfalls in local or production environments.

## Future improvements

- More documentation
  - Provide links
  - Describe simulator
  - Describe 'layer1' DHT
  - Describe ConnectionManager
  - Describe Manager-RpcLocal-RpcRemote pattern for components

### CODE REVIEW

File: packages/dht/README.md [hunk 1]
```diff
@@ -1,17 +1,99 @@
-# @streamr/dht (WIP)
+# @streamr/dht
 
-An implementation of the Kademlia-based Streamr DHT used for the trackerless network.
+A connectionful Kademlia based P2P distributed hash table (DHT). Implements an in-memory key-value data store where multiple creators of of data can list values behind a single key. The library is entirely browser compatible.
```
Comment (teogeb) [inline packages/dht/README.md]:
```suggestion
A connectionful Kademlia based P2P distributed hash table (DHT). Implements an in-memory key-value data store where multiple creators of data store values behind a single key. The library is entirely browser compatible.
```

File: packages/dht/README.md [hunk 2]
```diff
@@ -1,17 +1,99 @@
...
 
-## Generating Protobuf code
+All communication between peers utilizes protobuf, transported with the help of the custom `@streamr/proto-rpc` library.
```
Comment (teogeb) [inline packages/dht/README.md]:
```suggestion
All communication between peers utilizes [protobuf](https://protobuf.dev/), transported with the help of the custom `@streamr/proto-rpc` library.
```

File: packages/dht/README.md [hunk 3]
```diff
@@ -1,17 +1,99 @@
...
 
-Auto-generated protobuf/gRPC code is compiled during the build process.
-```bash
-  npm run build
+Connections in the DHT are established using WebSocket or WebRTC connections. The nodes will decide internally based on peers' connectivity information which connection types to use.
+
+The DHT also provides an interface for sending messages from peer to peer with the `DhtNode#send` function. The sent message is routed over the network to the target node's id.
```
Comment (teogeb) [inline packages/dht/README.md]:
```suggestion
The DHT also provides an interface for sending messages from peer to peer with the `DhtNode#send` function. The sent message is routed over the network to the target node.
```

Could also also briefly describe how the network topology is formed? I.e. nodes which have closest kademlia distance are neighbors, and how `randomPeers` is related to that? Are we always connected to all neighbors, or other nodes too?

File: packages/dht/README.md [hunk 4]
```diff
@@ -1,17 +1,99 @@
...
+
+## Running a node
+
+Running a bare DhtNode in the Streamr Network is heavily discouraged as it requires advanced configuration. However, if you do wish to run a node here are some recommended configurations.
+
+(the following values are examples for working production or test values check the streamr-client's default network configuration)
```
Comment (teogeb) [inline packages/dht/README.md]:
```suggestion
(the following values are examples; for working production or test values check the streamr-client's default network configuration)
```

File: packages/dht/README.md [hunk 5]
```diff
@@ -1,17 +1,99 @@
...
+  // Add a list of known entry points to the network
+  entryPoints: [{
+    kademliaId: new Uint8Array([1, 2, 3])
+  }], 
+  // a list of STUN and TURN servers. Critically important when opening webrtc conenctions behind NATs.
+  iceServers: [{
+    url: "stun.l.google.com",
+    port: 19302
+  }],
+  // A range of ports that the node will attempt to start a WebSocket server in. If you wish to use an exact port give equal values to min and max.
```
Comment (teogeb) [inline packages/dht/README.md]: It is also okay to start a node without a WebSocket server? What are consequences of that?

File: packages/dht/README.md [hunk 6]
```diff
@@ -1,17 +1,99 @@
...
+There two ways to configure TLS for a DhtNode's WS server.
+
+### Auto-certification
+
+When starting a node configure it to automatically fetch TLS certificates for a randomly generated domain name as such:
+
+```js
+const DhtNode = new DhtNode({
+  ...
+  websocketServerEnableTls: true,
```
Comment (teogeb) [inline packages/dht/README.md:50]: Not related to this PR: maybe `autoCertified* `config options could be moved into the `tlsCertificate` block in the future?

File: packages/dht/README.md [hunk 7]
```diff
@@ -1,17 +1,99 @@
...
+
+### local testing and develpment setup
+
+When running a node for testing or develpment a few configurations are important.
+
+```js
+const DhtNode = new DhtNode({
+  entryPoints: [], // Point this to your local dev entry point!
+  iceServers: [], // Keep empty to ensure webrtc connections are local
+  webrtcAllowPrivateAddresses: true, // Make sure that the value is true in local develpment
```
Comment (teogeb) [inline packages/dht/README.md:85]: What this does in practice? Allows nodes to connect directly to each others using LAN ip addresses or something?

**Review by teogeb — COMMENTED** [review]
This brief overview is already very helpful 👍

File: packages/dht/README.md (hunk 7 above)
Comment (juslesan) [reply packages/dht/README.md:85]: Yes, if this is false the nodes will filter out private/local ip addresses

File: packages/dht/README.md (hunk 3 above)
Comment (juslesan) [reply packages/dht/README.md]: Not necessarily important to detail that here. Technical details can be added in a separate section in the future

File: packages/dht/README.md (hunk 6 above)
Comment (juslesan) [reply packages/dht/README.md:50]: yeah at some point configuration related to specific features could be given their specific objects

**Review by teogeb — APPROVED** [review]

**Pull request merged**